frontend/frontend/src/views/PredatorPrey/PredatorPrey.vue

```diff
@@ -399,9 +430,17 @@ export default {
     if (this.$store.state.activeUser.isActive) { //Don't load presets if no one is logged in
       this.getAllPresets()
     }
-    //Set initial values, calling initialConditions computed property to be inherited by charts
-    this.N0 = this.simParamData[0]
-    this.P0 = this.simParamData[3]
+    //Set simulation params to default values
+    this.simParamData.length = 6 //Number of params in this model
+    this.simParamData[0] = this.defaultParams.N0
+    this.simParamData[1] = this.defaultParams.a
+    this.simParamData[2] = this.defaultParams.b
+    this.simParamData[3] = this.defaultParams.P0
+    this.simParamData[4] = this.defaultParams.c
+    this.simParamData[5] = this.defaultParams.d
```

Comment:
No change needed here just a comment - maybe an object would have been a better data structure to use since you could do this:

```suggestion
    this.simParamData.N0 = this.defaultParams.N0
    this.simParamData.a = this.defaultParams.a
    this.simParamData.b = this.defaultParams.b
    this.simParamData.P0 = this.defaultParams.P0
    this.simParamData.c = this.defaultParams.c
    this.simParamData.d = this.defaultParams.d
```

This would also get rid of loads of your magic numbers

Reply:
Makes the code more self descriptive

Reply:
Yeah i think you're right. I used an array as I knew for certain it was iterable within Vue (v-for) and javascript. I can see now that objects are also iterable in Vue, and can be in JS, so I'd use that in the future.